### ISSUE TITLE
Bold syntax highlighting broken for default syntax since 72d0220

### ISSUE THREAD

**Issue by brennen** [vimwiki/vimwiki#1325]
## description

cc: @tinmarino

Under the default syntax, highlighting is broken for bold `*` under some circumstances, at least since 72d0220. Reverting that commit seems to fix the worst breakage, although I do notice that an html-style comment still contains a bold where I don't expect one.

A couple of reproduction cases:

  - `*foo:* bar`
  - `<!-- exec-raw zsh -c 'grep --exclude-dir="*logscratch" -rino -- "- \[ \].*$" ~/notes/vimwiki' | perl -pe 's{^/home/brennen/notes/vimwiki/(.*?)\.wiki:(\d+):- \[ \] (.*?)$}{  - [[$1|$1]] - ($2) $3}' -->`

The second one of those looks complicated, but it should just be treated as a comment. Instead it's bold after the second `*` and stays that way until it encounters a `*` on another line halfway down the file.

Screenshots:

![2023-04-04-10:58:55](https://user-images.githubusercontent.com/74990/229868383-c5babdbd-d521-409e-b073-0ac002e977fd.png)
![2023-04-04-10:58:23](https://user-images.githubusercontent.com/74990/229868385-ff25f25e-6aa0-4ee0-8e26-fb2aa8999e35.png)

## settings / version

```
Version: 2022.12.02                                                                                              
Os: Linux                                                                                                        
Vim: 802                                                                                                         
Branch: dev                                                                                                      
Revision: dec6a9e                                                                                                
Date: 2023-03-22 01:22:25 -0300
```

Default syntax. See below for entire `vimrc`, although I'm guessing this is trivially reproducible.

### vimrc

Here's my [entire current `vimrc`](https://code.p1k3.com/gitea/brennen/bpb-kit/src/commit/bddc0b1da74177abc9894760f4ed0a87536c54cd/home/.vim/vimrc), but this can be reproduced with just the following:

```vim
" a stub vimrc for just running vimwiki on an otherwise stock setup

set nocompatible
set nomodeline
let mapleader = ","

filetype off

set rtp+=~/.vim/bundle/Vundle.vim/
call vundle#begin()

" let Vundle manage Vundle - required:
Plugin 'VundleVim/Vundle.vim'
Plugin 'vimwiki/vimwiki'

call vundle#end()
filetype plugin on
filetype indent on

" Turn on syntax
syntax on
```

**Comment by tinmarino** [comment]
@brennen, can you please check and assign back to me if the fix did not work?

**Comment by brennen** [comment]
Will do.

**Comment by Goli4thus** [comment]
@tinmarino, I've encountered this issue too on my end recently.
My workaround at the time was to append an additional `*` symbol.

For instance:
These had messed up bold highlight (before your fix; boldness would continue without end):
`*random text?*`
`*"random text"*`

"hacky workaround" was this (adding an additional `*`):
`*random text?**`
`*"random text"**`

Now with your fix in 7a48268, I had to remove my workaround to have proper bold highlighting (to the previous version described above).
I've just updated like 40 instances of this and each worked out fine.
IMO that's a good sign that your fix is working.
